src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -993,7 +993,7 @@ trait AppAndGroupFormats {
       // top-level ports fields are incompatible with IP/CT
       if (runSpec.ipAddress.isEmpty) {
         appJson = appJson ++ Json.obj(
-          "ports" -> runSpec.servicePorts,
+          "ports" -> runSpec.portDefinitions.map(_.port),
```

Comment:
This is a breaking change. Can you add more context to make sure that it makes sense?

If we decide to do this, then we should add a huge warning in our changelog and release notes.